File: LeetCode/657. 判断路线成圈.py
class Solution:
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        cntR=moves.count("R")
        cntL=moves.count("L")
        cntU=moves.count("U")
        cntD=moves.count("D")
        if cntR==cntL and cntU==cntD:
            return True
        else:
            return False

def stringToString(input):
    return input[1:-1].decode('string_escape')

def main():
    import sys
    def readlines():
        for line in sys.stdin:
            yield line.strip('\n')

    lines = readlines()
    while True:
        try:
            line = next(lines)
            moves = stringToString(line);
            
            ret = Solution().judgeCircle(moves)

            out = (ret);
            print(out)
        except StopIteration:
            break

if __name__ == '__main__':
    main()

File: LeetCode/657. 判断路线成圈.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
        int r=0,l=0,u=0,d=0;
        for(int i=0;i<moves.length();i++){
            char c=moves[i];
            if(c=='R') r++;
            else if(c=='L') l++;
            else if(c=='U') u++;
            else d++;
        }
        if(r==l && u==d) return true;
        else return false;
    }
};